### PR TITLE
launcher

### PR DESCRIPTION
**Pull request by matkt**
Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

## PR description

This PR adds a launcher to easily configure your client on the mainnet. 

You need to start besu with the `--Xlauncher` flag. Besu will ask questions and will generate a config file. If we restart with this flag and a config file already exists besu will not ask question.

If there is a config file and you want to force the launcher to ask question you need to start besu with `--Xlauncher-force` flag or remove the config file

We are created a library for that https://github.com/ConsenSys/quorum-mainnet-launcher 
With this dependencies https://github.com/ConsenSys/quorum-mainnet-launcher/blob/098df40acd7f7fb773f887f6096021312f97a08d/build.gradle#L93

## Sample 

```
? Which Ethereum network would you like to use ?
❯ mainnet
  rinkeby
  ropsten
  goerli
  dev
  classic
  kotti
  mordor
  yolo_v2
```

You can have more information with `--Xhelp` flag

```
Unstable options for Launcher
      --Xlauncher[=<isLauncherMode>]
         Activate the launcher if no configuration file is present.  (default:
           false)
      --Xlauncher-force[=<isLauncherModeForced>]
         Force to activate the launcher even if a configuration file is
           present.  (default: false)
```

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: besu/src/main/resources/org/hyperledger/besu/launcher/launcher.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
+[
+    {
+      "prompt-type" :"LIST",
+      "question":"Which ethereum network would you like to use ?",
```
Comment (timbeiko) [inline besu/src/main/resources/org/hyperledger/besu/launcher/launcher.json:4]:
```suggestion
      "question":"Which Ethereum network would you like to use ?",
```

Comment (timbeiko) [reply besu/src/main/resources/org/hyperledger/besu/launcher/launcher.json:4]: Maybe we should have an option to list them all?

File: besu/src/main/resources/org/hyperledger/besu/launcher/launcher.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                          "config-key": "rpc-http-apis",
+                          "available-options" : "org.hyperledger.besu.ethereum.api.jsonrpc.RpcApis$ALL_JSON_RPC_APIS"
+                        }
+                   ]
+               }
+            ]
+     } ,
+     {
+        "prompt-type" :"CONFIRM",
+        "question":"Do you want to enable P2P functionality ?",
```
Comment (timbeiko) [inline besu/src/main/resources/org/hyperledger/besu/launcher/launcher.json:61]: Is there a case where we _don't_ want this enabled?

Comment (matkt) [reply besu/src/main/resources/org/hyperledger/besu/launcher/launcher.json]: no it's true, I just removed it.

File: besu/src/main/resources/org/hyperledger/besu/launcher/launcher.json (hunk 1 above)
Comment (matkt) [reply besu/src/main/resources/org/hyperledger/besu/launcher/launcher.json]: Done

File: besu/src/main/java/org/hyperledger/besu/cli/DefaultCommandValues.java [hunk 3]
```diff
@@ -31,6 +31,7 @@
 import picocli.CommandLine;
 
 public interface DefaultCommandValues {
+
```
Comment (abdelhamidbakhta) [inline besu/src/main/java/org/hyperledger/besu/cli/DefaultCommandValues.java:34]: unnecessary change

File: besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java [hunk 4]
```diff
@@ -60,6 +60,7 @@ public ConfigOptionSearchAndRunHandler(
 
   private Optional<File> findConfigFile(
       final ParseResult parseResult, final CommandLine commandLine) {
+
```
Comment (abdelhamidbakhta) [inline besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java:63]: unnecessary change

File: besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java [hunk 5]
```diff
@@ -1515,6 +1515,22 @@ public void ethStatsContactOptionCannotBeUsedWithoutEthStatsServerProvided() {
             "The `--Xethstats-contact` requires ethstats server URL to be provided. Either remove --Xethstats-contact or provide an url (via --Xethstats=nodename:secret@host:port)");
   }
 
+  @Test
+  public void launcherDefautOptionAreValue() {
```
Comment (abdelhamidbakhta) [inline besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java:1519]: typo: Default instead of Defaut

**Review by abdelhamidbakhta — APPROVED** [review]
Looks good to me.

File: besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java (hunk 4 above)
Comment (matkt) [reply besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java:63]: Done

File: besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java (hunk 5 above)
Comment (matkt) [reply besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java:1519]: Done
